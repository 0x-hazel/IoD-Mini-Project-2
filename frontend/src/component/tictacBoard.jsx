import TicTac from "./ticTac";

// boardState: ('x' | 'o')[9]
// isSelectable: boolean
// changeTo: ('x' | 'o')
export function TicTacBoard({ boardState, isSelectable, changeTo }) {
  return (
    <div className="grid grid-rows-3 grid-cols-3">
      <div className="border-base-content border-r border-b">
        <TicTac />
      </div>
      <div className="border-base-content border-r border-l border-b">
        <TicTac />
      </div>
      <div className="border-base-content border-l border-b">
        <TicTac />
      </div>
      <div className="border-base-content border-t border-b border-r">
        <TicTac />
      </div>
      <div className="border-base-content border">
        <TicTac />
      </div>
      <div className="border-base-content border-t border-b border-l">
        <TicTac />
      </div>
      <div className="border-base-content border-t border-r">
        <TicTac />
      </div>
      <div className="border-base-content border-t border-r border-l">
        <TicTac />
      </div>
      <div className="border-base-content border-l border-t">
        <TicTac />
      </div>
    </div>
  );
}
