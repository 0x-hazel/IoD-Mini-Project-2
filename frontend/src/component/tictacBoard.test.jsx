import { expect, test } from "vitest";
import { TicTacBoard } from "./tictacBoard";

test("Snapshot test TicTacBoard", () => {
  expect(
    <TicTacBoard
      boardState={["", "x", "x", "", "o", "", "", "o", ""]}
      isSelectable={true}
      changeTo={() => {}}
    />
  ).toMatchSnapshot();
});
