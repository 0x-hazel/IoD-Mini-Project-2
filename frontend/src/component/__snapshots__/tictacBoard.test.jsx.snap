// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test TicTacBoard 1`] = `
<TicTacBoard
  boardState={
    [
      "",
      "x",
      "x",
      "",
      "o",
      "",
      "",
      "o",
      "",
    ]
  }
  changeTo={[Function]}
  isSelectable={true}
/>
`;
